components/location/popup.jsx

```diff
@@ -0,0 +1,319 @@
+// e.g. ['https://google.com']
+///////////
+function extractUrls(item) {
+  // web URL regex source: https://stackoverflow.com/questions/6927719/url-regex-does-not-work-in-javascript
+  const host_pattern = /(?:[a-z][\w-]+:(?:\/{1,3}|[a-z0-9%])|www\d{0,3}[.]|[a-z0-9.-]+[.][a-z]{2,4}\/)/;
+  const path_pattern = /(?:[^\s()<>]+|\(([^\s()<>]+|(\([^\s()<>]+\)))*\))+/;
+  const end_pattern = /(?:\(([^\s()<>]+|(\([^\s()<>]+\)))*\)|[^\s`!()[\]{};:'".,<>?«»“”‘’])/;
+  const web_pattern =
+    host_pattern.source + path_pattern.source + end_pattern.source;
+  // phone regexp adapted from: https://stackoverflow.com/a/16699507
+  const phone_pattern = /^(\+\d{1,2}\s)?\(?\d{3}\)?[\s.-]?\d{3}[\s.-]?\d{4}$/;
+  // email regexp adapted from https://www.regular-expressions.info/email.html
+  const email_pattern = /[\w.%+-]+@[\w.]+\.[a-z]{2,}/;
+  const url_pattern = new RegExp(
+    /\b/.source + // all URLs start after a word boundary
+      "(" +
+      web_pattern +
+      "|" +
+      phone_pattern.source +
+      "|" +
+      email_pattern.source +
+      ")",
+    "gi" // find all occurrences, case-insensitive
+  );
+
+  return item.match(url_pattern);
+}
+
+function parseLineBreaks(value) {
+  return value.replace(/\n/g, "<br />");
+}
+
+const PopupSection = ({ content, id }) => {
+  const { getTranslation } = useContext(LanguageContext);
+
+  return (
+    <div className="p row">
+      <p
+        // data-translation-id="${_.snakeCase(id)}"
+        className="key"
+      >
+        {getTranslation(id)}
+      </p>
+      {/** Kanad Note: I had to change this from a p tag to a div tag
+       * since it was causing issues when doing a ReactDOMServer render */}
+      <div className="value">{content}</div>
+    </div>
+  );
+};
+
+const NeedsMoney = ({ location }) => {
+  const { getTranslation } = useContext(LanguageContext);
+
+  let link = "";
+  if (location.seekingMoneyURL && location.seekingMoneyURL !== "") {
+    link = (
+      <a
+        data-translation-id="seeking_money_link"
+        href="${location.seekingMoneyURL}"
+        target="_blank"
+        // onclick="captureOutboundLink('${location.seekingMoneyURL}', 'donation')"
+      >
+        {getTranslation("seeking_money_link")}
+      </a>
+    );
+  }
+  return (
+    <span className="seekingMoney seeking-money card-badge">
+      <span data-translation-id="seeking_money">
+        {getTranslation("seeking_money")}
+      </span>{" "}
+      {link}
+    </span>
+  );
+};
+
+const OpenHours = ({ closingHours, id, openingHours }) => {
+  if (openingHours === "never" || openingHours === "not today") {
+    return <PopupSection content={openingHours} id={id} />;
+  } else {
+    const openHours = closingHours.map((closeHour, i) => {
+      <p>
+        <span>{openingHours[i]}</span> to <span>{closingHours[i]}</span>
+      </p>;
+    });
+    return <PopupSection content={openHours} id={id} />;
+  }
+};
+
+const NoIdNeeded = ({ location }) => {
```

Comment:
I think we talked about this briefly but some of these should probably move out if they're used in the location card as well

Reply:
Agreed!